app/auth/context/AuthContext.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { createContext, useContext, PropsWithChildren } from 'react';
+import { useStorageState } from '../hooks/useStorageState';
+import { router } from 'expo-router';
+import { DEV_HOST } from '../../../constants/urls';
+
+const API = process.env.EXPO_PUBLIC_API_URL || DEV_HOST;
+
+type AuthDataType = {
+  email: string;
+  accessToken: string;
+};
+
+type AuthContextType = {
+  user: AuthDataType | null;
+  signIn: (email: string | null, password: string | null) => Promise<void>;
+  signUp: (
+    email: string | null,
+    username: string | null,
+    password: string | null,
+  ) => Promise<void>;
+  signOut: () => Promise<void>;
+  isLoading: boolean;
+};
+
+const AuthContext = createContext<AuthContextType | undefined>(undefined);
+
+export const useAuth = () => {
+  const value = useContext(AuthContext);
+  if (!value) {
+    throw new Error('useAuth must be wrapped in an <AuthProvider />');
+  }
+
+  return {
+    ...value,
+  };
+};
+
+export const AuthProvider = ({ children }: PropsWithChildren) => {
+  const [[isLoading, authData], setAuthData] = useStorageState('auth') as [
+    [boolean, AuthDataType | null],
+    (value: AuthDataType | null) => void,
+  ];
+
+  const signIn = async (email: string | null, password: string | null) => {
+    if (!email || !password) {
+      throw new Error('email and password are required');
+    }
+
+    try {
+      const response = await fetch(`${API}/auth/login`, {
```

Comment:
I still see this path and some others here and other files that could use a constant, but let's do it in a followup.